onnxruntime/core/optimizer/transpose_optimizer/transpose_optimizer.cc

```diff
@@ -469,6 +496,14 @@ static void TransposeInput(api::GraphRef& graph, api::NodeRef& node, size_t i,
 
   // Case 1: input is a constant with a known list of consumer nodes
   if (constant != nullptr && consumers->comprehensive) {
+    // This is a special case where the constant is 1D with length == perm.
+    // TODO: TransposeInitializer should be updated to handle this case.
+    // Permute1DConstant permutes the constant and adds a new initializer. The old initializer is only removed if
+    // there are no other consumers.
+    if (constant->Shape().size() == 1 && constant->Shape()[0] == gsl::narrow_cast<int64_t>(perm.size())) {
```

Comment:
If `constant->Shape()[0] == 0` (i.e. it's a scalar) can we return here instead of falling through?

Reply:
we can add this check here... this case is usually seen in resize op( because of the existing bug in the op spec version 11) we already have a special handling for resize in layout transformer where we do not call transpose input for empty initializer